lib/Date/Utility.pm

```diff
@@ -501,7 +499,7 @@ sub _parse_datetime_param {
         $year += ($year <= 30) ? 2000 : 1900;
     }
 
-    my $epoch = timegm_nocheck($second, $minute, $hour, $day, $month - 1, $year);
+    my $epoch = timegm($second, $minute, $hour, $day, $month - 1, $year);
```

Comment:
But simply changing it back to `timegm_nocheck` breaks one of the new tests. So, there is a bit more work to do.